sphinx/config.py

```diff
@@ -286,6 +281,12 @@ def __init__(self, dirname, filename, overrides, tags):
         self.setup = config.get('setup', None)
```

Comment:
The `if 'extensions' in overrides:` check should come before `self.extensions = config.get('extensions', [])`